### PR TITLE
rework dates and update outdated documentation

### LINKED ISSUE #77: sod with wrong description
Hello, there is a little error [here](https://taskwarrior.org/docs/named_dates.html). In the table and in the image sod is describted for today but I believe it is actually tomorrow.

### LINKED ISSUE #97: Setting a task to be due:sow sets due date at start of current week
According to the documentation, sow should set a task to be due at the start of the coming week:

image

However, I tried using this command and the due-date instead was set to be the start of the current week! Also, note that `socw` is apparently not recognized.

I am running taskwarrior on OSX, installed using Brew!

```
~ $ task version

task 2.5.3 built for Darwin
Copyright (C) 2006 - 2021 P. Beckingham, F. Hernandez.

Taskwarrior may be copied only under the terms of the MIT license, which may be found in the Taskwarrior source kit.

Documentation for Taskwarrior can be found using 'man task', 'man taskrc', 'man task-color', 'man task-sync' or at http://taskwarrior.org

~ $ task add test due:socw
'socw' is not a valid date in the 'Y-M-D' format.
~ $ task add test due:sow
Created task 17.
~ $ task description:test
[task next ( description:test )]

ID Age  Due   Description Urg
17 8s   -2d   test          10

1 task
```

Interestingly, the same issue can not be reproduced on a raspberry pi, version 2.5.1:

```
~ $ task add test due:sow
Created task 5.
~ $ task
[task next]

ID Age Tag     Due  Description                                                         Urg
 5 -           3d   test                                                                 7.3

5 tasks
~ $ task version

task 2.5.1 built for linux
Copyright (C) 2006 - 2016 P. Beckingham, F. Hernandez.

Taskwarrior may be copied only under the terms of the MIT license, which may be found in the Taskwarrior source kit.

Documentation for Taskwarrior can be found using 'man task', 'man taskrc', 'man task-color', 'man task-sync' or at http://taskwarrior.org
```

### PR DESCRIPTION
**Pull request by felixschurk**
Closes #77. Closes #97.

Closes https://github.com/GothenburgBitFactory/taskwarrior/issues/1956.
Closes https://github.com/GothenburgBitFactory/taskwarrior/issues/3275.
Closes https://github.com/GothenburgBitFactory/taskwarrior/issues/3005.
Closes https://github.com/GothenburgBitFactory/taskwarrior/issues/3559.

In general, there is quite a huge mismatch between the actual behavior and the documented one. This PR tries to clean it up.

### CODE REVIEW

File: content/docs/named_dates.md [hunk 1]
```diff
@@ -47,38 +46,55 @@ For this, use a precise date format.
 The months of the year are interpreted as the first day of the next month of that name in the future.
 To specify a month in the past, use a different date format.
 
-* `jan`, `january`   - The date of the nearest future January 1st, at 0:00:00 local.
+* `jan`, `january`, `Janurary`    - The date of the nearest future January 1st, at 0:00:00 local.
```
Comment (djmitche) [inline content/docs/named_dates.md:49]:
```suggestion
* `jan`, `january`, `January`    - The date of the nearest future January 1st, at 0:00:00 local.
```

File: content/docs/dates.md [hunk 2]
```diff
@@ -138,26 +138,24 @@ Here is the full set:
...
-* `tomorrow`                   - Local date for tomorrow, with time 00:00:00. Same as `sod`.
+* `tomorrow`                   - Local date for tomorrow, with time 00:00:00.
 * `monday`, `tuesday`, ...     - Local date for the specified day, after today, with time 00:00:00. Can be shortened, e.g. `mon`, `tue` {{< label >}}2.6.0{{< /label >}} Can be capitalized, e.g. `Monday`, `Tue`
 * `january`, `february`, ...   - Local date for the specified month, 1st day, with time 00:00:00. Can be shortened, e.g. `jan`, `feb`. {{< label >}}2.6.0{{< /label >}} Can be capitalized, e.g. `January`, `Feb`.
-* `later`, `someday`           - Local 2038-01-18, with time 00:00:00. A date far away, with semantically meaningful to GTD users.
-* `soy`                        - Local date for the next year, January 1st, with time 00:00:00.
-* `eoy`                        - Local date for this year, December 31st, with time 00:00:00.
-* `soq`                        - Local date for the start of the next quarter (January, April, July, October), 1st, with time 00:00:00.
+* `later`, `someday`           - Local 9999-12-30, with time 00:00:00. A date far away, with semantically meaningful to GTD users.
```
Comment (djmitche) [inline content/docs/dates.md:147]:
```suggestion
* `later`, `someday`           - Local 9999-12-30, with time 00:00:00. A date far away, with a meaning to GTD users.
```

File: content/docs/named_dates.md [hunk 3]
```diff
@@ -47,38 +46,55 @@ For this, use a precise date format.
...
-
-[![](../../images/year.png)](../../images/year.png)
+Bellow figure illustrates some of the cases.
+
+```goat
+                            sopm     NOW
+                              |       |
+                              v       v
++-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+
+| DEC | Jan | Feb | Mar | Apr | May | Jun | Jul | Aug | Sep | Oct | Nov | Dec | Jan |
```
Comment (djmitche) [inline content/docs/named_dates.md:77]:
```suggestion
| Dec | Jan | Feb | Mar | Apr | May | Jun | Jul | Aug | Sep | Oct | Nov | Dec | Jan |
```

**Review by djmitche — APPROVED** [review]
Minor typo fixes. I love the diagrams!

**Comment by djmitche** [comment]
(also sorry this took me so long..)

**Comment by felixschurk** [comment]
No worries, I was just also not sure if someone is "managing" this repo at all :D

**Comment by djmitche** [comment]
I think we're all in it together :)

**Pull request merged**